kitsune/search/v2/search.py

```diff
@@ -0,0 +1,209 @@
+import bleach
+from datetime import datetime
+from abc import ABC, abstractmethod
+
+from elasticsearch_dsl import Search as DSLSearch, Q
+from kitsune.search.v2.es7_utils import es7_client
+from kitsune.search.config import WIKI_DOCUMENT_INDEX_NAME, QUESTION_INDEX_NAME
+
+
+class BaseConfig(ABC):
+    """Semi-abstract base class for Search configs."""
+
+    @property
+    @abstractmethod
+    def index(self):
+        pass
+
+    @property
+    @abstractmethod
+    def fields(self):
+        """Array of fields to search over."""
+        pass
+
+    @property
+    @abstractmethod
+    def highlight_fields(self):
+        """Array of fields to highlight."""
+        pass
+
+    @property
+    def filter(self):
+        """Filter which returns all documents of this type to search over."""
+        return Q("match", _index=self.index)
+
+    @abstractmethod
+    def product_filter(self, product):
+        """Filter which returns documents with the correct product."""
+        pass
+
+    @abstractmethod
+    def make_result(self, hit):
+        """Transform a hit into a result dictionary."""
+        pass
+
+    @staticmethod
+    def extract_highlight(hit):
+        highlight = getattr(hit.meta, "highlight", None)
+        if highlight:
+            return bleach.clean(
+                next(iter(highlight.to_dict().values()))[0],
+                tags=["strong"],
+                strip=True,
+            )
+        return None
+
+
+class AAQConfig(BaseConfig):
```

Comment:
Shall we rename this to `AAQSearch` or something within this context? Having the word config here implies that it's about configuration but there's also a method that builds the result for the AAQ.